### PR TITLE
Improve performance for sa_autowrite for large data

### PR DESCRIPTION
**Pull request by MaiMee1**
### Fixes
- Increase performance by at least 2/3
- Add function to individually insert data `insert_xsv_data`
  - Specify rows to start and stop reading
  - Report rows read and committed
- Add function to read IMDb LIST files

### Changes
- New package structure
- Update `sa_autowrite/README.md`
- Remove dependency `stringcase`
- Auto-generated models added with user edits
- New module in utils: `utils/iter.py` (petition to start renaming utils modules for brevity)
- Auto-generated package now has `models = [...]` for convenience

### Depreciation warning
- Will depreciate `insert_data()` soon
- `parse_line_iter()` is depreciated. Do not use.

### Tests
- Part of the tests for `parse_line_re()` used to parse each line in IMDb LIST files.

### CODE REVIEW

**Comment by MaiMee1** [comment]
@NutthanichN @th-bunratta Wanna review? If not, to speed things up I'll merge it soon

File: sa_autowrite/parsers.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    lst = []
+    for line in get_data_lines(filename, encoding=encoding):
+        try:
+            dct = parse_line_re(line)
+        except ValueError:
+            if line.startswith('---'):
+                # last line
+                return
+            else:
+                raise
```
Comment (th-bunratta) [inline sa_autowrite/parsers.py:227]: What does one of these lines raise?

File: sa_autowrite/insert.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                        else:
+                            raise
+                instance = model(**datab)
+                instances.append(instance)
+                counter += 1
+                if counter % 1000 == 0:
+                    session.add_all(instances)
+                    session.commit()
+                    instances = []
+                    print(f'Commited {counter}')
```
Comment (th-bunratta) [inline sa_autowrite/insert.py:176]: Committed?

File: sa_autowrite/insert.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+
+# def read_in_chunks(file_object, chunk_size=1024):
+#     """Lazy function (generator) to read a file piece by piece.
+#     Default chunk size: 1k."""
+#     while True:
+#         data = file_object.read(chunk_size)
+#         if not data:
+#             break
+#         yield data
```
Comment (th-bunratta) [inline sa_autowrite/insert.py:24]: Why are these lines commented out?

**Review by th-bunratta — APPROVED** [review]
Looks good, but they still left me some doubts I had of some portions of code.

**Review by NutthanichN — APPROVED** [review]

File: sa_autowrite/parsers.py (hunk 1 above)
Comment (MaiMee1) [reply sa_autowrite/parsers.py:227]: The ValueError catched in line 222 raised from `parse_line_re()` from not matching the regex

File: sa_autowrite/insert.py (hunk 2 above)
Comment (MaiMee1) [reply sa_autowrite/insert.py:176]: You're right! That function is depreciated though.

File: sa_autowrite/insert.py (hunk 3 above)
Comment (MaiMee1) [reply sa_autowrite/insert.py:24]: Those are a template from stack overflow someone wrote. I kept it there for if we want to refactor the code in the future.

File: sa_autowrite/insert.py (hunk 2 above)
Comment (MaiMee1) [reply sa_autowrite/insert.py]: Solved in 9b55662

**Comment by MaiMee1** [comment]
Ok. I'll merge it now.

**Pull request merged**